lint.js

```diff
@@ -78,45 +46,48 @@ ${colors.reset + error.message}
     console.log(output);
 }
 
-const main = (str, callback) => {
-    options.openapi = yaml.safeLoad(str,{json:true});
-    resolver.resolve(options)
-    .then(function() {
-        options.status = 'resolved';
-        callback(options);
-    })
-    .catch(function(err) {
-        options.status = 'rejected';
-        console.warn(err);
-        process.exit(0);
-    });
-};
 
-const command = (file, cmd) => {
+const readOrError = file => {
+    try {
+        return loader.loadSpec(file, { resolve: true });
+    }
+    catch (error) {
+        if (error.name == 'OpenError') {
+            console.error('Could not open file: ' + error.message);
+        }
+        else if (error.name == 'ReadError') {
+            console.error('Could not read YAML/JSON from file: ' + error.message);
+        }
+        else {
+            console.error(error);
+        }
+        process.exit(1);
+    }
+}
+
+const command = async (file, cmd) => {
+  const spec = await readOrError(file);
+  const options = { openapi: spec };
 
   linter.loadRules(cmd.rules, cmd.skip);
 
-  if (file && file.startsWith('http')) {
-      console.log('GET ' + file);
-      fetch(file, {agent:options.agent}).then(function (res) {
-          if (res.status !== 200) throw new Error(`Received status code ${res.status}`);
-          return res.text();
-      }).then(function (body) {
-          main(body, lintResolvedSchema);
-      }).catch(function (err) {
-          console.warn(err);
-      });
-  }
-  else {
-      fs.readFile(file,'utf8',function(err,data){
-          if (err) {
-              console.warn(err);
-              return;
-          }
-          main(data, lintResolvedSchema);
-      });
-  }
-
+  validator.validate(options.openapi, options, function(err, options) {
+      if (err) {
+          console.log(colors.red + 'Specification schema is invalid.' + colors.reset);
+          formatSchemaError(err, options.context);
+          process.exit(1);
```

Comment:
is it ok to exit here? shouldn't throw a validation error or something and be caught in the main func?

Reply:
Agreed. There should be some error output for the exit IMO

Reply:
Hey, this was confusing as the formatSchemaError was outputting content, so there _is_ always content. I've made those functions return strings, and its a bit more clear now. 

```
          const output = formatSchemaError(err, options.context);
          console.error(output);
          process.exit(1);
```